lib/matplotlib/artist.py

```diff
@@ -236,15 +235,16 @@ def stale(self):
 
     @stale.setter
     def stale(self, val):
-        # only trigger call-back stack on being marked as 'stale'
-        # when not already stale
-        # the draw process will take care of propagating the cleaning
-        # process
-        if not (self._stale == val):
-            self._stale = val
-            # only trigger propagation if marking as stale
-            if self._stale:
-                self.pchanged()
+        self._stale = val
+
+        # if the artist is animated it does not take normal part in the
+        # draw stack and is not expected to be drawn as part of the normal
+        # draw loop (when not saving) so do not propagate this change
+        if self.get_animated():
+            return
```

Comment:
I will revert the set change when i get back to my normal computer.

On Wed, Aug 12, 2015, 11:43 AM Thomas Caswell <tcaswell@gmail.com> wrote:

> The stale handler does not propagate on animated artists, but you can
> still set all of their properties.  If you are using blitting you are still
> responsible for doing the redraw. If you are not blitting then animated
> needs to be false so they get drawn by the normal draw method which means
> stale state does propagate, but draw_idle takes care of collecting those.
>
> I am a bit worried about saving, i suspect that should also be checked in
> the figure call back installed by pyplot.
>
> On Wed, Aug 12, 2015, 11:33 AM Benjamin Root <notifications@github.com>
> wrote:
>
>> In lib/matplotlib/artist.py
>> <https://github.com/matplotlib/matplotlib/pull/4738#discussion_r36874315>
>> :
>>
>> > -        # only trigger call-back stack on being marked as 'stale'
>> > -        # when not already stale
>> > -        # the draw process will take care of propagating the cleaning
>> > -        # process
>> > -        if not (self._stale == val):
>> > -            self._stale = val
>> > -            # only trigger propagation if marking as stale
>> > -            if self._stale:
>> > -                self.pchanged()
>> > +        self._stale = val
>> > +
>> > +        # if the artist is animated it does not take normal part in the
>> > +        # draw stack and is not expected to be drawn as part of the normal
>> > +        # draw loop (when not saving) so do not propagate this change
>> > +        if self.get_animated():
>> > +            return
>>
>> Is it still going to be possible to change the properties of an animated
>> object? I typically do things like changing the alpha or color as part of
>> an animation.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/matplotlib/matplotlib/pull/4738/files#r36874315>.
>>
>